lib/active_merchant/billing/gateways/xpay.rb

```diff
@@ -30,77 +30,162 @@ def initialize(options = {})
+        add_auth_purchase_params(post, amount, payment_method, options)
+        commit(:preauth, post, options)
       end
 
       def capture(amount, authorization, options = {})
         post = {}
-        commit('capture', post, options)
+        commit(:capture, post, options)
       end
 
       def void(authorization, options = {})
         post = {}
-        commit('void', post, options)
+        commit(:void, post, options)
       end
 
       def refund(amount, authorization, options = {})
         post = {}
-        commit('refund', post, options)
+        commit(:refund, post, options)
       end
 
       def verify(credit_card, options = {})
         post = {}
-        commit('verify', post, options)
+        add_invoice(post, 0, options)
+        add_credit_card(post, credit_card)
+        commit(:verify, post, options)
       end
 
       def supports_scrubbing?
         true
       end
 
-      def scrub(transcript) end
+      def scrub(transcript)
+        transcript.
+          gsub(%r((X-Api-Key: )(\w|-)+), '\1[FILTERED]').
+          gsub(%r(("pan\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r(("cvv\\?":\\?")\d+), '\1[FILTERED]')
+      end
 
       private
 
-      def add_invoice(post, money, options) end
+      def add_invoice(post, amount, options)
+        currency = options[:currency] || currency(amount)
+        post[:order] = {
+          orderId: options[:order_id],
+          amount: localized_amount(amount, currency),
+          currency: currency
+        }.compact
+      end
+
+      def add_credit_card(post, payment_method)
+        post[:card] = {
+          pan: payment_method.number,
+          expiryDate: expdate(payment_method),
+          cvv: payment_method.verification_value
+        }
+      end
+
+      def add_payment_method(post, payment_method)
+        add_credit_card(post, payment_method) if payment_method.is_a?(CreditCard)
+      end
+
+      def add_customer_data(post, payment_method, options)
+        post[:order][:customerInfo] = {
+          cardHolderName: payment_method.name,
+          cardHolderEmail: options[:email]
+        }.compact
+      end
+
+      def add_address(post, options)
+        if options[:billing_address] || options[:address]
+          post[:order][:customerInfo][:billingAddress] = {
+            name: address[:name],
+            street: address[:address1],
+            additionalInfo: address[:address2],
+            city: address[:city],
+            postCode: address[:zip],
+            country: address[:country]
+          }.compact
+        end
+        if options[:shipping_address]
+          post[:order][:customerInfo][:shippingAddress] = {
+            name: address[:name],
+            street: address[:address1],
+            additionalInfo: address[:address2],
+            city: address[:city],
+            postCode: address[:zip],
+            country: address[:country]
+          }.compact
+        end
+      end
+
```

Comment:
💬 Comment:

Let's reduce the implementation scope and leave `add_recurrence` and `add_exemptions` to the time those features get requested